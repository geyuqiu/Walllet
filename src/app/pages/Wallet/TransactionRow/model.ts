export interface Transaction {
	id: string;
	amount: string;
	fee: string;
	sender: string;
	recipient: string;
	timestamp: string;
};
