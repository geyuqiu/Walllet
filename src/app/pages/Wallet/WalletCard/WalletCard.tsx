import {Divider} from "app/components/Divider/Divider";
import {Dropdown} from "app/components/Dropdown/Dropdown";
import {Icon} from "app/components/Icon/Icon";
import React from "react";
import {useHistory} from "react-router-dom";
import styled from 'styled-components';

import {SvgCollection} from '../../../assets/svg';
import {Button} from '../../../components/Button/Button';
import {Card} from '../../../components/Card/Card';
import {Wallet} from '../model';
import {hideTextMiddle} from '../TransactionRow/TransactionRow';

type LogoContainerProps = {
	width: number;
}

const LogoContainer = styled.div<LogoContainerProps>`
  ${({width}) => width && `
    width: ${width}px;
  `};
  height: auto;
`;

const {Logo} = SvgCollection;

type WalletProps = {
	wallets: Wallet[];
	wallet: Wallet | null;
}

export const WalletCard = ({wallets, wallet}: WalletProps) => {
	const history = useHistory();
	const walletDisplayOptions = [
		{label: 'all', value: "all"},
		{label: 'favorites', value: "favorites"},
		{label: 'ledger', value: "ledger"},
	];

	return (
		<>
			{!wallet && <p>Loading ...</p>}
			{wallet &&
			<Card className="bg-green-dark sm:flex sm:justify-center">
				<Card className="bg-black-darkest text-white rounded-lg w-full max-w-screen-xl flex justify-center flex-col lg:flex-row">
					<div className="flex items-center my-auto cursor-pointer" onClick={() => history.push('fee')}>
						<LogoContainer width={57}>
							<Logo width={44}/>
						</LogoContainer>

						<span className="text-2xl font-bold lg:hidden xl:block" data-testid="logo__text">ARK Wallet</span>
					</div>
					<div className="lg:hidden">
						<Divider className="border-black-light dark:border-theme-secondary-600" type="horizontal"/>
					</div>
					<div className="lg:contents hidden">
						<Divider className="border-black-light dark:border-theme-secondary-600" type="vertical"/>
					</div>
					<Dropdown
						toggleContent={
							<div className="relative">
								<Button size="icon" className="text-left" icon="Address" fill='#000000' stroke='#FBC457'>
									<span className="hidden sm:block">{wallet.address}</span>
									<span className="block sm:hidden">{hideTextMiddle(wallet.address, 7, 8)}</span>
									<Icon name="ChevronDown" width={20} height={20}/>
								</Button>
							</div>
						}
						options={walletDisplayOptions}
						dropdownClass="top-3 text-left"
					/>
					<div className="hidden lg:block">
						<Divider className="border-black-light dark:border-theme-secondary-600" type="vertical"/>
					</div>
					<div>
						<Icon name="Balance" width={20} height={20} fill='#000000' stroke='#FBC457'/>
						{wallet.balance}
					</div>
				</Card>
			</Card>
			}
		</>
	);
};
