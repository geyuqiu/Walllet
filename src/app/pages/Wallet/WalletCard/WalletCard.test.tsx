import {render} from "@testing-library/react";
import React from "react";

import {Wallet} from '../model';
import {buildLabelAndValue, WalletCard} from "./WalletCard";

describe("WalletCard", () => {
	it("should match snapshot", () => {
		const wallets = [
			{
				address: "D6Z26L69gdk9qYmTv5uzk3uGepigtHY4ax",
				balance: "8498932552758785"
			},
			{
				address: "DEyaFhDuaoQyKbFH4gJtYZvKkB6umyrEUj",
				balance: "2000044816999638"
			},
		];

		const { container } = render(<WalletCard  wallet={{
			address: 'AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW',
			balance: '0'
		}} wallets={wallets}/>);

		expect(container).toMatchSnapshot();
	});

	it("wallet is null", () => {
		const {container} = render(<WalletCard wallet={null} wallets={[]}/>);

		expect(container).toMatchSnapshot();
	});

	it("buildLabelAndValue", () => {
		const wallets: Wallet[] = [
			{address: 'address1', balance: 'balance1'},
			{address: 'address2', balance: 'balance2'},
			{address: 'address3', balance: 'balance3'},
			{address: 'address4', balance: 'balance4'},
			{address: 'address5', balance: 'balance5'},
		]

		expect(buildLabelAndValue(wallets)).toEqual([
			{label: 'address1', value: 'address1'},
			{label: 'address2', value: 'address2'},
			{label: 'address3', value: 'address3'},
			{label: 'address4', value: 'address4'},
			{label: 'address5', value: 'address5'},
		]);
	});
});
