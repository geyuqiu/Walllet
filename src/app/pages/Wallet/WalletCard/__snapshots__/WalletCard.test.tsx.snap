// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletCard should match snapshot 1`] = `
<div>
  <div
    class="Cardstyles__Wrapper-xnuhyv-0 iAsJMU bg-green-dark sm:flex sm:justify-center"
    data-testid="Card"
  >
    <div
      class="Cardstyles__Wrapper-xnuhyv-0 iAsJMU bg-black-darkest text-white rounded-lg w-full max-w-screen-xl flex justify-center lg:items-center flex-col lg:flex-row"
      data-testid="Card"
    >
      <div
        class="flex items-center my-auto cursor-pointer"
      >
        <div
          class="sc-gsTCUz dXshFa"
          width="57"
        >
          <svg
            width="44"
          >
            logo.svg
          </svg>
        </div>
        <span
          class="text-2xl font-bold lg:hidden xl:block"
          data-testid="logo__text"
        >
          ARK Wallet
        </span>
      </div>
      <div
        class="lg:hidden"
      >
        <div
          class="Divider-sc-10xryrv-0 ijsPwK border-black-light dark:border-theme-secondary-600"
          type="horizontal"
        />
      </div>
      <div
        class="lg:contents hidden"
      >
        <div
          class="Divider-sc-10xryrv-0 dCquGe border-black-light dark:border-theme-secondary-600"
          type="vertical"
        />
      </div>
      <div
        class="relative border border-gray-darkest border-opacity-10 rounded-l-3xl rounded-r-3xl"
      >
        <span
          data-testid="dropdown__toggle"
        >
          <div
            class="flex justify-between items-center h-11"
          >
            <div
              class="flex items-center"
            >
              <div
                class="hidden sm:block"
              >
                <div
                  class="Circle__CircleWrapper-sc-1hq8kv9-0 cqSuMy rounded-l-3xl border-r bg-black-light ring-theme-background"
                >
                  <div
                    class="sc-bdfBwQ dfWOLJ"
                    height="15"
                    width="15"
                  >
                    <svg
                      fill="#000000"
                      stroke="#FBC457"
                    >
                      address.svg
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="ml-5"
              >
                <span
                  class="hidden sm:block lg:hidden xl:block"
                >
                  AdzbhuDTyhnfAqepZzVcVsgd1Ym6FgETuW
                </span>
                <span
                  class="block sm:hidden"
                >
                  AdzbhuD...m6FgETuW
                </span>
                <span
                  class="hidden lg:block xl:hidden"
                >
                  AdzbhuDTyhn...1Ym6FgETuW
                </span>
              </div>
            </div>
            <div
              class="sc-bdfBwQ dfWOLJ mx-5"
              height="15"
              width="15"
            >
              <svg>
                chevron-down.svg
              </svg>
            </div>
          </div>
        </span>
      </div>
      <div
        class="hidden lg:contents"
      >
        <div
          class="Divider-sc-10xryrv-0 dCquGe border-black-light dark:border-theme-secondary-600"
          type="vertical"
        />
      </div>
      <div
        class="flex flex-row mt-6 lg:mt-0"
      >
        <div
          class="hidden sm:block"
        >
          <div
            class="Circle__CircleWrapper-sc-1hq8kv9-0 cqSuMy border rounded-full mr-3 ring-theme-background"
          >
            <div
              class="sc-bdfBwQ dfWOLJ"
              height="15"
              width="15"
            >
              <svg
                fill="#000000"
                stroke="#FBC457"
              >
                balance.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          class="flex flex-col"
        >
          <span
            class="text-gray-darkest"
          >
            Balance
          </span>
          <span
            class="font-bold"
            data-testid="balance__Amount"
          >
            0 DARK
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WalletCard wallet is null 1`] = `
<div>
  <p>
    Loading ...
  </p>
</div>
`;
